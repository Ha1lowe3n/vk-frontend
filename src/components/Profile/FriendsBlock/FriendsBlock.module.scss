.friendsBlock {
    .friends {
        padding: 6px 10px 8px;
        display: flex;
        flex-wrap: wrap;
        justify-content: space-between;
    }
}
