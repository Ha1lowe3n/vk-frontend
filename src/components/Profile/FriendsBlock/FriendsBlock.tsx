import clsx from "clsx";
import React from "react";
import { Divider } from "../../WhiteBlock/Divider/Divider";
import { WhiteBlock } from "../../WhiteBlock/WhiteBlock";
import { WhiteBlockInfo } from "../../WhiteBlock/WhiteBlockInfo/WhiteBlockInfo";

import styles from "./FriendsBlock.module.scss";
import { Person } from "./Person/Person";

const dataPersons = [
    {
        src: "https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1",
        name: "Vladilen",
    },
    {
        src: "https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1",
        name: "Vladilen",
    },
    {
        src: "https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1",
        name: "Vladilen",
    },
    {
        src: "https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1",
        name: "Vladilen",
    },
    {
        src: "https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1",
        name: "Vladilen",
    },
    {
        src: "https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1",
        name: "Vladilen",
    },
];

export const FriendsBlock: React.FC = () => {
    return (
        <WhiteBlock className={clsx("page_block", styles.friendsBlock)}>
            <WhiteBlockInfo title={"Friends"} count={100} />
            <div className={styles.friends}>
                {dataPersons.map((person, i) => (
                    <Person key={i} src={person.src} name={person.name} />
                ))}
            </div>

            <Divider />

            <WhiteBlockInfo title={"Friends online"} count={15} />
            <div className={styles.friends}>
                {dataPersons.map((person, i) => (
                    <Person key={i} src={person.src} name={person.name} />
                ))}
            </div>
        </WhiteBlock>
    );
};
