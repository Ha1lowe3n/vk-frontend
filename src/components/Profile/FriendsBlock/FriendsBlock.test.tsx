import { render, screen } from "@testing-library/react";
import { FriendsBlock } from "./FriendsBlock";

it("render FriendsBlock component", () => {
    render(<FriendsBlock />);
    expect(screen.getAllByAltText("person avatar")[0]).toBeInTheDocument();
    expect(screen.getByText("Friends")).toBeInTheDocument();
    expect(screen.getByText("Friends online")).toBeInTheDocument();
});

it("FriendsBlock snapshot", () => {
    const view = render(<FriendsBlock />);
    expect(view).toMatchSnapshot();
});
