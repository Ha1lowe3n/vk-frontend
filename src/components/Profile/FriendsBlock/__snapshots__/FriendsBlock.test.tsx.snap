// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FriendsBlock snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="whiteBlock page_block friendsBlock"
      >
        <div
          class="whiteBlockInfo"
        >
          <span
            class="title"
          >
            Friends
          </span>
          <span
            class="count"
          >
            100
          </span>
        </div>
        <div
          class="friends"
        >
          <div
            class="person"
          >
            <a
              class="imgRow"
              href="#"
            >
              <img
                alt="person avatar"
                src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
              />
            </a>
            <a
              class="link_text"
              href="#"
            >
              Vladilen
            </a>
          </div>
          <div
            class="person"
          >
            <a
              class="imgRow"
              href="#"
            >
              <img
                alt="person avatar"
                src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
              />
            </a>
            <a
              class="link_text"
              href="#"
            >
              Vladilen
            </a>
          </div>
          <div
            class="person"
          >
            <a
              class="imgRow"
              href="#"
            >
              <img
                alt="person avatar"
                src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
              />
            </a>
            <a
              class="link_text"
              href="#"
            >
              Vladilen
            </a>
          </div>
          <div
            class="person"
          >
            <a
              class="imgRow"
              href="#"
            >
              <img
                alt="person avatar"
                src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
              />
            </a>
            <a
              class="link_text"
              href="#"
            >
              Vladilen
            </a>
          </div>
          <div
            class="person"
          >
            <a
              class="imgRow"
              href="#"
            >
              <img
                alt="person avatar"
                src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
              />
            </a>
            <a
              class="link_text"
              href="#"
            >
              Vladilen
            </a>
          </div>
          <div
            class="person"
          >
            <a
              class="imgRow"
              href="#"
            >
              <img
                alt="person avatar"
                src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
              />
            </a>
            <a
              class="link_text"
              href="#"
            >
              Vladilen
            </a>
          </div>
        </div>
        <div
          style="border: 0.5px solid #d8dde3; background-color: rgb(216, 221, 227);"
        />
        <div
          class="whiteBlockInfo"
        >
          <span
            class="title"
          >
            Friends online
          </span>
          <span
            class="count"
          >
            15
          </span>
        </div>
        <div
          class="friends"
        >
          <div
            class="person"
          >
            <a
              class="imgRow"
              href="#"
            >
              <img
                alt="person avatar"
                src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
              />
            </a>
            <a
              class="link_text"
              href="#"
            >
              Vladilen
            </a>
          </div>
          <div
            class="person"
          >
            <a
              class="imgRow"
              href="#"
            >
              <img
                alt="person avatar"
                src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
              />
            </a>
            <a
              class="link_text"
              href="#"
            >
              Vladilen
            </a>
          </div>
          <div
            class="person"
          >
            <a
              class="imgRow"
              href="#"
            >
              <img
                alt="person avatar"
                src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
              />
            </a>
            <a
              class="link_text"
              href="#"
            >
              Vladilen
            </a>
          </div>
          <div
            class="person"
          >
            <a
              class="imgRow"
              href="#"
            >
              <img
                alt="person avatar"
                src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
              />
            </a>
            <a
              class="link_text"
              href="#"
            >
              Vladilen
            </a>
          </div>
          <div
            class="person"
          >
            <a
              class="imgRow"
              href="#"
            >
              <img
                alt="person avatar"
                src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
              />
            </a>
            <a
              class="link_text"
              href="#"
            >
              Vladilen
            </a>
          </div>
          <div
            class="person"
          >
            <a
              class="imgRow"
              href="#"
            >
              <img
                alt="person avatar"
                src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
              />
            </a>
            <a
              class="link_text"
              href="#"
            >
              Vladilen
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="whiteBlock page_block friendsBlock"
    >
      <div
        class="whiteBlockInfo"
      >
        <span
          class="title"
        >
          Friends
        </span>
        <span
          class="count"
        >
          100
        </span>
      </div>
      <div
        class="friends"
      >
        <div
          class="person"
        >
          <a
            class="imgRow"
            href="#"
          >
            <img
              alt="person avatar"
              src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
            />
          </a>
          <a
            class="link_text"
            href="#"
          >
            Vladilen
          </a>
        </div>
        <div
          class="person"
        >
          <a
            class="imgRow"
            href="#"
          >
            <img
              alt="person avatar"
              src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
            />
          </a>
          <a
            class="link_text"
            href="#"
          >
            Vladilen
          </a>
        </div>
        <div
          class="person"
        >
          <a
            class="imgRow"
            href="#"
          >
            <img
              alt="person avatar"
              src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
            />
          </a>
          <a
            class="link_text"
            href="#"
          >
            Vladilen
          </a>
        </div>
        <div
          class="person"
        >
          <a
            class="imgRow"
            href="#"
          >
            <img
              alt="person avatar"
              src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
            />
          </a>
          <a
            class="link_text"
            href="#"
          >
            Vladilen
          </a>
        </div>
        <div
          class="person"
        >
          <a
            class="imgRow"
            href="#"
          >
            <img
              alt="person avatar"
              src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
            />
          </a>
          <a
            class="link_text"
            href="#"
          >
            Vladilen
          </a>
        </div>
        <div
          class="person"
        >
          <a
            class="imgRow"
            href="#"
          >
            <img
              alt="person avatar"
              src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
            />
          </a>
          <a
            class="link_text"
            href="#"
          >
            Vladilen
          </a>
        </div>
      </div>
      <div
        style="border: 0.5px solid #d8dde3; background-color: rgb(216, 221, 227);"
      />
      <div
        class="whiteBlockInfo"
      >
        <span
          class="title"
        >
          Friends online
        </span>
        <span
          class="count"
        >
          15
        </span>
      </div>
      <div
        class="friends"
      >
        <div
          class="person"
        >
          <a
            class="imgRow"
            href="#"
          >
            <img
              alt="person avatar"
              src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
            />
          </a>
          <a
            class="link_text"
            href="#"
          >
            Vladilen
          </a>
        </div>
        <div
          class="person"
        >
          <a
            class="imgRow"
            href="#"
          >
            <img
              alt="person avatar"
              src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
            />
          </a>
          <a
            class="link_text"
            href="#"
          >
            Vladilen
          </a>
        </div>
        <div
          class="person"
        >
          <a
            class="imgRow"
            href="#"
          >
            <img
              alt="person avatar"
              src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
            />
          </a>
          <a
            class="link_text"
            href="#"
          >
            Vladilen
          </a>
        </div>
        <div
          class="person"
        >
          <a
            class="imgRow"
            href="#"
          >
            <img
              alt="person avatar"
              src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
            />
          </a>
          <a
            class="link_text"
            href="#"
          >
            Vladilen
          </a>
        </div>
        <div
          class="person"
        >
          <a
            class="imgRow"
            href="#"
          >
            <img
              alt="person avatar"
              src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
            />
          </a>
          <a
            class="link_text"
            href="#"
          >
            Vladilen
          </a>
        </div>
        <div
          class="person"
        >
          <a
            class="imgRow"
            href="#"
          >
            <img
              alt="person avatar"
              src="https://sun1-55.userapi.com/s/v1/ig2/4AmTw3rdunMJaSLZppSACbdzwsYy-w5fhHszb-fvHFLvv9ea2fe5yrcJE8K_3BsCsJv5MAlWYJRwyIpsB_5SCN9J.jpg?size=50x50&quality=96&crop=0,145,989,989&ava=1"
            />
          </a>
          <a
            class="link_text"
            href="#"
          >
            Vladilen
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
